File: server-laravel-hybrid/app/Models/PersonalProject.php
<?php

namespace App\Models;

use Rhino\Models\RhinoModel;

/**
 * PersonalProject — user-owned model for the `personal` route group.
 *
 * Owned by user_id (no organization). Read/write isolation is enforced by
 * PersonalProjectScope (auto-discovered global scope). user_id is auto-stamped
 * from the authenticated user on create.
 */
class PersonalProject extends RhinoModel
{
    protected $table = 'personal_projects';

    protected static function booted(): void
    {
        static::creating(function (PersonalProject $project) {
            if (auth()->check() && !$project->user_id) {
                $project->user_id = auth()->id();
            }
        });
    }

    protected $fillable = [
        'user_id',
        'title',
        'description',
        'status',
    ];

    protected $validationRules = [
        'title' => 'required|string|max:255',
        'description' => 'nullable|string',
        'status' => 'required|string|max:255',
    ];

    public static $allowedFilters = ['title', 'status'];
    public static $allowedSorts = ['title', 'status'];
    public static $defaultSort = 'created_at';
    public static $allowedFields = ['id', 'user_id', 'title', 'description', 'status', 'created_at', 'updated_at'];
    public static $allowedIncludes = ['user'];

    public function user(): \Illuminate\Database\Eloquent\Relations\BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}

File: server-rails-hybrid/app/models/personal_project.rb
# frozen_string_literal: true

# PersonalProject — user-owned model for the `personal` route group.
#
# Owned by user_id (no organization). Read/write isolation is enforced by
# Scopes::PersonalProjectScope (auto-discovered default scope). user_id is
# auto-stamped from the authenticated user on create. Served only by the
# `personal` group; the org-owned Project table serves agency/vendor.
class PersonalProject < Rhino::RhinoModel
  include Discard::Model

  rhino_filters :title, :status
  rhino_sorts :title, :status
  rhino_default_sort "created_at"
  rhino_fields :id, :user_id, :title, :description, :status, :created_at, :updated_at
  rhino_includes :user

  validates :title, length: { maximum: 255 }, allow_nil: true
  validates :status, inclusion: { in: %w[draft active completed archived] }, allow_nil: true

  belongs_to :user

  before_validation :auto_set_user_id, on: :create

  private

  def auto_set_user_id
    return if user_id.present?
    return unless defined?(RequestStore)

    current_user = RequestStore.store[:rhino_current_user]
    self.user_id = current_user.id if current_user
  end
end
